### PR TITLE
Document reasoning for why CustomMultiChildLayout size can't depend on children and how to do it

### LINKED ISSUE #18426: Document reasoning for why CustomMultiChildLayout size can't depend on children and how to do it
After reading the CustomMultiChildLayout and associated docs, it's not very clear why it auto-grabs the largest constraint sizes instead of passing the incoming constraints to its delegate and letting the delegate return a size for itself. 

It could be a separate widget but still useful to have rather than having a custom render object to avoid all the boilerplate of doing non-layout related things.

cc @matthew-carroll

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
## Description

The title is taken from https://github.com/flutter/flutter/issues/18426 because this PR does exactly what the issue is title (and fixes a typo).  

I will most likely write an article about how to use `MultiChildRenderObject` as part of the Flutter Clock challenge, but I suppose that it would be against the guidelines to link it in the docs anyway - so I hope that mentioning relevant classes, even though they lack documentation, is sufficient for now. I think that I remember learning about it as quite easy by example when I used it for the first time, which is why I also mentioned `Stack`, even though there might be easier examples because `Stack` is fairly complex.

## Related Issues

Resolves #18426.

## Tests

No tests are needed for documentation only.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Comment by creativecreatorormaybenot** [comment]
cc @xster @Hixie

File: packages/flutter/lib/src/rendering/custom_layout.dart
Comment (Piinks) [inline packages/flutter/lib/src/rendering/custom_layout.dart]:
```suggestion
/// If you want to build a custom layout where you define the size of that widget
```

**Review by Piinks — COMMENTED** [review]
Hey @creativecreatorormaybenot, thanks for this! 
I don't know that this explains _why_, but it does provide direction for someone who is looking to do this. 
I just noted a typo below.

**Comment by creativecreatorormaybenot** [comment]
@Piinks Thank you for taking a look! I added the explanation to the documentation now, I missed that before, so thanks for pointing it out.

I paraphrased what @hixie and @xster commented in #18426 for the first part btw.

File: packages/flutter/lib/src/rendering/custom_layout.dart
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: Maybe simplify this long sentence a little: "... properties of the children. This was a design decision to simplify the delegate implementations: This way, the delegate implementations don't have to also handle various intrinsic sizing functions if the parent's size depended on the children."

**Review by Piinks — APPROVED** [review]
LGTM modulo @goderbauer's comments.

**Comment by dnfield** [comment]
@fkorotkov - this Pull request didn't get any status updates from Cirrus for some reason.  See for example:

```
query LabeledPullRequestsWithReviews {
  repository(owner: "flutter", name: "flutter") {
    pullRequest(number: 47797) {
      id
      number
      mergeable
      commits(last: 100) {
        nodes {
          commit {
            abbreviatedOid
            oid
            committedDate
            pushedDate
            status {
              state
              contexts {
                state
                targetUrl
                description
                context
              }
            }
          }
        }
      }
    }
  }
}
```

In the GitHub graphQL explorer.  Other PRs are getting these statuses.  Any idea what happened here?

**Comment by fkorotkov** [comment]
@dnfield I think it's because of the use of Checks API since #43097. You need to query for a CheckSuite but these objects are not shown in the explorer according to [documentation](https://developer.github.com/v4/object/checksuite/). But with `Accept: application/vnd.github.antiope-preview+json` header you'll be able to get `Commit.checkSuites` and something like this should work:

```graphql
query LabeledPullRequestsWithReviews {
  repository(owner: "flutter", name: "flutter") {
    pullRequest(number: 47797) {
      id
      number
      mergeable
      commits(last: 100) {
        nodes {
          commit {
            checkSuites {
              id
              status
              app {
                name
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by dnfield** [comment]
@fkorotkov - ahh I thought I was still seeing statuses on others that are also recent, but I was wrong.  Thanks for the tip!

**Comment by Piinks** [comment]
@creativecreatorormaybenot This change was reverted after it was accidentally merged in yesterday. It looks like there was an analyzer error caused by trailing white space. Would you like to open a new pull request to re-land this change?

**Comment by creativecreatorormaybenot** [comment]
@Piinks I am sorry for that - see #48451.

**Comment by Piinks** [comment]
No worries at all! It was not your fault. :) I'll review the new PR so we can get it back in.

**Pull request merged**
